[Bug]: Donut chart is buggy flickering while hover over the pie during the animation of filling in

### Application/Team

Data AI

### What happened?

A donut chart has an animation of filling in the circle of the chart when one of the pies (sectors) was removed from that chart.
A bug happened when we mouse over any of the pies which are left in the chart but during the animation described above (filling in the circle of the chart).

Example of reproducing the bug.
https://github.com/carbon-design-system/carbon-charts/assets/71228700/30a2efd9-f666-42a5-91c2-7d72c41752a5

One of the possible reasons could be the conflict of two animations which occurs simultaneously: 
- increase in width of the pie on mouse over;
- filling in the chart circle

### Version

@carbon/charts-react@1.14.2
and
@carbon/charts@0.58.2
@carbon/charts-react@0.58.0

### Data & options used

```js
const data = [
  {
    group: 'L22I P66EP L22I P66EP L22I P66EP',
    value: 65000,
  },
  {
    group: 'JQAI 2M4L1',
    value: 75000,
  },
];

const options = {
  resizable: true,
  donut: {
    center: {
      label: 'Browsers',
    },
  },
  height: '400px',
  toolbar: {
    enabled: false,
  },
  legend: {
    enabled: false,
  },
};
```


### Relevant log output

```js
Error: <path> attribute d: Expected arc flag ('0' or '1'), "…8.676378695001,138.676909299001,…".
eval @ index.js:1
attrTween.js:15 Error: <path> attribute d: Expected arc flag ('0' or '1'), "…8.676378695001,138.676909299001,…".
eval @ attrTween.js:15
Show 1 more frame
Show less
 Error: <path> attribute d: Expected arc flag ('0' or '1'), "…45505024500255,138.4571760090025…".
eval @ index.js:1
attrTween.js:15 Error: <path> attribute d: Expected arc flag ('0' or '1'), "…45505024500255,138.4571760090025…".
eval @ attrTween.js:15
Show 1 more frame
Show less
 Error: <path> attribute d: Expected arc flag ('0' or '1'), "….5190781550392,136.5351568710389…".
eval @ index.js:1
attrTween.js:15 Error: <path> attribute d: Expected arc flag ('0' or '1'), "….5190781550392,136.5351568710389…".
eval @ attrTween.js:15
Show 1 more frame
Show less
...
etc.
```


### StackBlitz example

https://stackblitz.com/edit/react-imtvrq?file=src%2Findex.js,src%2Fdata.js,package.json

### What priority level would this be in your opinion?

P2

--- a/packages/core/src/components/graphs/pie.ts
+++ b/packages/core/src/components/graphs/pie.ts
@@ -1,4 +1,5 @@
 import { arc, interpolate, pie, select } from 'd3'
+import { debounce } from 'lodash-es'
 import { convertValueToPercentage, getProperty } from '@/tools'
 import { pie as pieConfigs } from '@/configuration'
 import { Component } from '@/components/component'
@@ -25,6 +26,7 @@ function arcTween(a: any, arcFunc: any) {
 export class Pie extends Component {
 	type = 'pie'
 	renderType = RenderTypes.SVG
+	isRendering = false
 
 	// We need to store our arcs
 	// So that addEventListeners()
@@ -53,6 +55,8 @@ export class Pie extends Component {
 		const options = this.getOptions()
 		const { groupMapsTo } = options.data
 		const { valueMapsTo } = options.pie
+
+		this.isRendering = true
 
 		// remove any slices that are valued at 0 because they dont need to be rendered and will create extra padding
 		const displayData = this.model.getDisplayData().filter((data: any) => data[valueMapsTo] > 0)
@@ -127,6 +131,10 @@ export class Pie extends Component {
 			// Tween
 			.attrTween('d', function (a: any) {
 				return arcTween.bind(this)(a, self.arc)
+			})
+			.end()
+			.finally(() => {
+				self.isRendering = false
 			})
 
 		// Draw the slice labels
@@ -383,11 +391,9 @@ export class Pie extends Component {
 
 	addEventListeners() {
 		const self = this
-		this.parent
-			.selectAll('path.slice')
-			.on('mouseover', function (event: MouseEvent, datum: any) {
-				const hoveredElement = select(this)
 
+		const debouncedMouseOver = debounce((hoveredElement: any) => {
+			if (!this.isRendering) {
 				hoveredElement
 					.classed('hovered', true)
 					.transition('pie_slice_mouseover')
@@ -398,6 +404,29 @@ export class Pie extends Component {
 						})
 					)
 					.attr('d', self.hoverArc)
+			}
+		}, 120, { trailing: true })
+
+		const debouncedMouseOut = debounce((hoveredElement: any) => {
+			if (!this.isRendering) {
+				hoveredElement
+					.classed('hovered', false)
+					.transition('pie_slice_mouseout')
+					.call((t: any) =>
+						self.services.transitions.setupTransition({
+							transition: t,
+							name: 'pie_slice_mouseout'
+						})
+					)
+					.attr('d', self.arc)
+			}
+		}, 120, { trailing: true })
+
+		this.parent
+			.selectAll('path.slice')
+			.on('mouseover', function (event: MouseEvent, datum: any) {
+				const hoveredElement = select(this)
+				debouncedMouseOver(hoveredElement)
 
 				// Dispatch mouse event
 				self.services.events.dispatchEvent(Events.Pie.SLICE_MOUSEOVER, {
@@ -445,16 +474,7 @@ export class Pie extends Component {
 			})
 			.on('mouseout', function (event: MouseEvent, datum: any) {
 				const hoveredElement = select(this)
-				hoveredElement
-					.classed('hovered', false)
-					.transition('pie_slice_mouseout')
-					.call((t: any) =>
-						self.services.transitions.setupTransition({
-							transition: t,
-							name: 'pie_slice_mouseout'
-						})
-					)
-					.attr('d', self.arc)
+				debouncedMouseOut(hoveredElement)
 
 				// Dispatch mouse event
 				self.services.events.dispatchEvent(Events.Pie.SLICE_MOUSEOUT, {

--- a/packages/core/src/components/graphs/pie.ts
+++ b/packages/core/src/components/graphs/pie.ts
@@ -1,5 +1,5 @@
 import { arc, interpolate, pie, select } from 'd3'
-import { debounce } from 'lodash-es'
+import { delay } from 'lodash-es'
 import { convertValueToPercentage, getProperty } from '@/tools'
 import { pie as pieConfigs } from '@/configuration'
 import { Component } from '@/components/component'
@@ -392,41 +392,22 @@ export class Pie extends Component {
 	addEventListeners() {
 		const self = this
 
-		const debouncedMouseOver = debounce((hoveredElement: any) => {
-			if (!this.isRendering) {
-				hoveredElement
-					.classed('hovered', true)
-					.transition('pie_slice_mouseover')
-					.call((t: any) =>
-						self.services.transitions.setupTransition({
-							transition: t,
-							name: 'pie_slice_mouseover'
-						})
-					)
-					.attr('d', self.hoverArc)
-			}
-		}, 120, { trailing: true })
-
-		const debouncedMouseOut = debounce((hoveredElement: any) => {
-			if (!this.isRendering) {
-				hoveredElement
-					.classed('hovered', false)
-					.transition('pie_slice_mouseout')
-					.call((t: any) =>
-						self.services.transitions.setupTransition({
-							transition: t,
-							name: 'pie_slice_mouseout'
-						})
-					)
-					.attr('d', self.arc)
-			}
-		}, 120, { trailing: true })
-
 		this.parent
 			.selectAll('path.slice')
 			.on('mouseover', function (event: MouseEvent, datum: any) {
 				const hoveredElement = select(this)
-				debouncedMouseOver(hoveredElement)
+				if (!self.isRendering) {
+					hoveredElement
+						.classed('hovered', true)
+						.transition('pie_slice_mouseover')
+						.call((t: any) =>
+							self.services.transitions.setupTransition({
+								transition: t,
+								name: 'pie_slice_mouseover'
+							})
+						)
+						.attr('d', self.hoverArc)
+				}
 
 				// Dispatch mouse event
 				self.services.events.dispatchEvent(Events.Pie.SLICE_MOUSEOVER, {
@@ -474,7 +455,20 @@ export class Pie extends Component {
 			})
 			.on('mouseout', function (event: MouseEvent, datum: any) {
 				const hoveredElement = select(this)
-				debouncedMouseOut(hoveredElement)
+				delay(() => {
+					if (!self.isRendering) {
+						hoveredElement
+							.classed('hovered', false)
+							.transition('pie_slice_mouseout')
+							.call((t: any) =>
+								self.services.transitions.setupTransition({
+									transition: t,
+									name: 'pie_slice_mouseout'
+								})
+							)
+							.attr('d', self.arc)
+					}
+				}, 100)
 
 				// Dispatch mouse event
 				self.services.events.dispatchEvent(Events.Pie.SLICE_MOUSEOUT, {

--- a/packages/core/src/configuration-non-customizable.ts
+++ b/packages/core/src/configuration-non-customizable.ts
@@ -241,6 +241,9 @@ export const transitions = {
 	pie_slice_mouseover: {
 		duration: 100
 	},
+	pie_slice_mouseout: {
+		duration: 100
+	},
 	pie_chart_titles: {
 		duration: 375
 	},

